pkgs/top-level/all-packages.nix

```diff
@@ -20461,7 +20461,7 @@ with pkgs;
 
   ### SCIENCE/MATH
 
-  almonds = pythonPackages.callPackage ../applications/science/math/almonds { };
+  almonds = callPackage ../applications/science/math/almonds { };
```

Comment:
Why not use `python3Packages.callPackage`?

Reply:
Using that is discouraged because that way, you lose the ability to override the python package set. Overriding a single dependency is not always what you want because you cannot use multiple versions of the same python package.

Reply:
But the same applies to the `pkgs` package set. Using `.override` on a derivation only overrides the immediate deps, not the whole set. If anybody wanted to override the whole set, they can use overlays.

Reply:
But for Python applications, it's common to require a specific version of a dependency. This can be done by overriding `pythonPackages` inside the application's expression.